### PR TITLE
ENH: stats.skew: add array-API support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Follow-up to gh-20292

#### What does this implement/fix?
This adds array-API support to `scipy.stats.skew`. It might be useful to review one commit at a time:
- 788b54aa38cd25262d7a1e5bd34b762379785a4c removes code that is no longer used (`nan_policy` support is handled by the `_axis_nan_policy` decorator)
- bb3575679665d9ac6a6b359679b935f8a2eebd15 attempts to simplify some of the existing pure-NumPy code, making it look a little closer to what the array-API requires
- 0d33968873314f9fb5dfe6bc8f41a50d5edd7608 drafts the conversion to array API
- 69d0c8ab9b6acbf0e7eeafc20b99f0a539b8979d tests with array-API backends (and fixes some things in the implementation)
- 0b7317de0071a2db507d828903fdec0486b5ea4c (unrelated to the main point of this PR, but needs to get taken care of) Makes the `_axis_nan_policy` decorator raise when it gets arguments that won't currently work with the array API. It only works when these parameters are passed as keyword arguments (which they *should* be anyway), but I am not too concerned because a) array-API support is still experimental and behind an environment variable and b) these features should be available with non-NumPy backends by the time the environment variable is removed, at which point this error will be removed. Nothing like this commit will be needed with other array-API conversions.

#### Additional information
After this, I will open an issue with a list of stats functions I think are ripe for array API support. We can point to this PR as an example and hopefully crowdsource the effort. I'd be willing to review PRs for functions on that list.

### CODE REVIEW

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -3381,31 +3381,37 @@ def test_moment_array_api(self, xp, order, axis, center):
...
-class TestSkewKurtosis:
+class SkewKurtosisTest:
     scalar_testcase = 4.
     testcase = [1., 2., 3., 4.]
     testmathworks = [1.165, 0.6268, 0.0751, 0.3516, -0.6965]
 
+
+class TestSkew(SkewKurtosisTest):
     def test_empty_1d(self):
+        # This is not essential behavior to maintain w/ array API
```
Comment (lucascolley) [inline scipy/stats/tests/test_stats.py:3392]: perhaps more clear to `@skip_xp_backends(np_only=True)`, although I don't have a strong opinion. Right now, not parameterising with `xp` very clearly means 'this is just for NumPy'. In the future, when we potentially have the majority of tests converted, that might not be so clear. But happy with this for now, can be left for the future.

**Review by tupui — APPROVED** [review]
LGTM, I had a look at the individual commits as suggested and it seems right to me. Also the changes on the tests look good.

I was just wondering if we might want to move the first `if` logic into the array API check function. But that's another topic anyway.

**Review by lucascolley — COMMENTED** [review]
agreed, beat me to it @tupui :) just a small typo and one thought about the testing

File: scipy/stats/tests/test_stats.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/tests/test_stats.py:3392]: I will consider talking this on to my next PR. But honestly this test can probably just be removed. We've talked about silencing this warning throughout stats.

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -1121,6 +1121,8 @@ def _var(x, axis=0, ddof=0, mean=None):
 @_axis_nan_policy_factory(
     lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
 )
+# nan_policy handled by `_axis_nan_policy, but needs to be left
```
[the remark this answers is not in the document]
Comment (mdhaber) [reply scipy/stats/_stats_py.py:1124]: Oops I'll fix this next time.

**Pull request merged**